### DISCUSSION TITLE
Preload images

### DISCUSSION THREAD

**Discussion by Arctomachine** [vercel/next.js#37385] [Help] [answered]
Lazy loading images is nice. But there is one major issue with this method.
No matter how good they are optimized and how fast internet connection is, there is bound to be a delay before image appears. And the way they appear isnt exactly pleasant. On slower connection images may pop up when they are already half way above top viewport border without user even knowing.
One of solutions people suggest every time for this question is to use eager loading or priority. It may look like a solution, but in reality it is not. In order for it to work it have to be applied to each and every meaningful big image on page. It in turn starts loading all big images at the same time at initial page load, how many of them there are. Which essentially ruins all effort for quickest delivery of page.

What I think could be a solution is to still use lazy load, but load images ahead of time, at certain range from current scroll position. This way image will already be loaded when user scrolls to its first pixel. But how can it be done? Are there native methods to do so? Can it be done by fiddling with wrapper component?

**Comment by icyJoseph** [comment, marked as answer]
There's an intrinsic problem with network IO, and that is that you cannot predict the time it will take to finish. 

The native lazy loading used by browsers uses heuristics to decide when it is best lazy load images. https://web.dev/browser-level-image-lazy-loading/#distance-from-viewport-thresholds

> Note that the people who made next/image are the same authors from the web.dev article above.

The lazy loading used by Next.js uses a bunch of JavaScript to figure out the best time to fetch them, and depending on the props you use it might even use the native lazy loading. It also makes sure to use native lazy loading to cover for cases when JavaScript is disabled. So I believe next/image is doing quite a lot for us developers. 

Perhaps you mean something like this prop, https://nextjs.org/docs/api-reference/next/image#lazyboundary, which helps you to tell each image when it is sensible to start loading. Again, if the user scrolls very fast, then even the best lazy boundary and techniques will fall short.

**Reply by Arctomachine** [reply]
I must have overlooked this parameter when reading about component. Default 200px is definitely not enough for any remote connection, so I set it to 150% and now it works much better.

**Reply by jsefiani** [reply]
It looks like the `lazyBoundary` was removed. Has anyone found a good solution for preloading images? This is especially needed for image carousels to provide a good UX.

**Comment by ricardotech** [comment]
Nowadays which one is the best approach to pre-load images in Next.js?

**Comment by Atharvk7** [comment]
I think we can also use SSR to load images fatser in ReactJS
